src/hyperlane-adapter/HyperlaneSenderAdapterV2.sol

```diff
@@ -112,7 +121,7 @@ contract HyperlaneSenderAdapter is ISingleMessageDispatcher, IBatchedMessageDisp
     MessageLib.Message[] memory _messages = new MessageLib.Message[](1);
     _messages[0] = MessageLib.Message({ to: _to, data: _data });
 
-    bytes memory payload = abi.encode(_messages, msgId, block.chainid, msg.sender);
+    bytes memory payload = EncodeDecodeUtil.encode(_messages, msgId, block.chainid, msg.sender);
```

Comment:
Used the library I created for encoding/decoding to encode the messages with extra relevant data to generate a payload